gc/src/lib.rs

```diff
@@ -34,9 +36,7 @@ pub use gc::force_collect;
 ///
 /// See the [module level documentation](./) for more details.
 pub struct Gc<T: Trace + ?Sized + 'static> {
-    // XXX We can probably take advantage of alignment to store this
-    root: Cell<bool>,
-    ptr: Shared<GcBox<T>>,
+    ptr_root: Cell<NonZero<*const GcBox<T>>>,
```

Comment:
We should make sure to also have the PhantomData here that used to be in Shared<T>